### PR TITLE
Convert fixes

### PR DESCRIPTION
**Pull request by mikapfl**
I found a couple of assorted problems while working on my PR, which I propose to fix here.

For the example updates / fixes, I re-ran all examples in the docstrings I found and fixed problems where they wouldn't run and updated the output.

The other fix is just a small change which increases test coverage somewhat because the wrong data type was being tested.

### CODE REVIEW

File: pint_xarray/tests/test_conversion.py [hunk 1]
```diff
@@ -358,6 +358,8 @@ def test_convert_units(self, type, variant, units, error, match):
         )
 
         if error is not None:
+            if type == "DataArray":
+                obj = obj["a"]
```
Comment (keewis) [inline pint_xarray/tests/test_conversion.py:362]: could you move this up two lines and remove the assignment below?

**Review by keewis — COMMENTED** [review]
oh, yes, most of these examples were written before we introduced `_repr_inline_`.

I think it would be good to have a CI that checks the doctests (using a github action), so we don't regress again. I will merge this first and then add that CI.

File: pint_xarray/tests/test_conversion.py (hunk 1 above)
Comment (mikapfl) [reply pint_xarray/tests/test_conversion.py]: sure

**Comment by keewis** [comment]
thanks. If you want, you can also add an entry to `whats-new.rst`, the doctests at least have been broken since before the release of 0.1 so this is user visible.

**Comment by mikapfl** [comment]
I added a small changelog entry. :slightly_smiling_face:

**Comment by keewis** [comment]
thanks, @mikapfl

**Pull request merged**